src/tools/idlc/src/plugin.c

```diff
@@ -64,6 +72,59 @@ static void *loadsym(void *handle, const char *symbol)
 #endif
 }
 
+#define SUBPROCESS_PIPE_MEMORY_LIMIT 1024 * 1024
+static int run_library_locator(const char *command, char **out_output, int *out_return_code) {
+  size_t output_size = 128;
+  char *output = (char*) malloc(output_size);
+  size_t output_pt = 0;
+  FILE *pipe;
+  int ret = 0;
+  int c;
+
+  if (!output) {
+    // out-of-memory
+    return -1;
+  }
```

Comment:
I'd set `output` to NULL and the `output_size` to `0` so there's only one place where allocation is done (in the loop). I'm not against this, but it does make things slightly simpler.